.github/ansible/redeploy-derper.yml

```
@@ -0,0 +1,25 @@
+---
+- name: Update derper node
+  hosts: derper
+  become: yes
+
+  tasks:
+  - name: Make sure derper is stopped
+    ansible.builtin.systemd:
+      state: stopped
+      enabled: yes
+      name: derper
+  - name: Fetch derper binary
+    get_url:
```

they recommend
```suggestion
    ansible.builtin.get_url:
```
apparently

---

So I can't really find how atomically this writes to the destination.  Which seems such a basic question which is not answered.  But what happens when the download fails halfway?  Is the original file left untouched?

Also, how does that work for the entire flow here?  Is the restart not happening in that case and this service is left stopped?

---

Yes a failed step would leave the derper stopped in this case.

Also from the docs
`unsafe_writes`
```
By default this module uses atomic operations to prevent data corruption or inconsistent reads from the target files, but sometimes systems are configured or just broken in ways that prevent this. One example is docker mounted files, which cannot be updated atomically from inside the container and can only be written in an unsafe manner.
```

---

yeah, i also found `unsafe_writes` which i interpreted as a very weak hint that by default they try and do the write atomically.  but this isn't described at all :(

Indeed, let's try it and assume it only replaces the file atomically if the download succeeds.  It'd just be nice to document these kind of things...